kore/test/Test/Kore/Step/Simplification.hs

```diff
@@ -63,23 +69,35 @@ import Kore.Step.Simplification.Data
     , SimplifierT
     )
 import qualified Kore.Step.Simplification.Data as Kore
+import Kore.Step.SMT.Lemma
+    ( declareSMTLemmas
+    )
+import Kore.Syntax.Module
+    ( ModuleName (ModuleName)
+    )
 import Logic
     ( LogicT
     )
 import SMT
     ( NoSMT
     )
-
 import qualified Test.SMT as Test
 
 runSimplifier :: Env Simplifier -> Simplifier a -> IO a
-runSimplifier env = Test.runSMT . Kore.runSimplifier env
+runSimplifier env = Test.runSMT userInit . Kore.runSimplifier env
+  where
+    userInit =
```

Comment:
I think this should really be:

```suggestion
    userInit = SMT.AST.declare (smtDeclarations metadataTools)
```
as [here](https://github.com/kframework/kore/blob/master/kore/test/Test/Kore/Strategies/Common.hs#L86). You will need to remove the latter.